app/controllers/api/v2/roles_controller.rb

```diff
@@ -51,11 +51,30 @@ def destroy
         process_response @role.destroy
       end
 
+      api :POST, "/roles/:id/clone", N_("Clone a role")
+      param :id, String, :required => true
+      param :new_name, String, :required => true, :desc => N_('Name of the cloned role')
```

Comment:
I changed to `:new_name`, which is correct one. Interestingly enough hammer sends both `:name` and `:new_name` with the same value, that is why I haven't discovered it:

![](http://i.imgur.com/Gsq0OIF.png)

I can see the value of adding different description to a newly cloned role, but how setting organizations should work is not obvious. Should new clone have the organizations from original when you do not supply any orgs, or should it have none? I believe it would be better to use update for any modifications.